cmd/thanos/partition_tenants_test.go

```diff
@@ -0,0 +1,290 @@
+				{TenantName: "tenant2", Weight: 20},
+			},
+			expectedTenants:     2,
+			validateAllAssigned: true,
+		},
+		{
+			name:                "empty tenant list",
+			numShards:           3,
+			tenantWeights:       []TenantWeight{},
+			expectedTenants:     0,
+			validateAllAssigned: true,
+		},
+		{
+			name:      "equal weights",
+			numShards: 2,
+			tenantWeights: []TenantWeight{
+				{TenantName: "tenant1", Weight: 10},
+				{TenantName: "tenant2", Weight: 10},
+				{TenantName: "tenant3", Weight: 10},
+				{TenantName: "tenant4", Weight: 10},
+			},
+			expectedTenants:     4,
+			validateAllAssigned: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			buckets, weights := computeTenantAssignments(tt.numShards, tt.tenantWeights)
+
+			// Verify all shards exist in the maps
+			if len(buckets) != tt.numShards {
+				t.Errorf("expected %d buckets, got %d", tt.numShards, len(buckets))
+			}
+			if len(weights) != tt.numShards {
+				t.Errorf("expected %d weight entries, got %d", tt.numShards, len(weights))
+			}
+
+			// Count total assigned tenants
+			totalAssigned := 0
+			assignedTenants := make(map[string]bool)
+			for i := 0; i < tt.numShards; i++ {
+				if _, ok := buckets[i]; !ok {
+					t.Errorf("bucket %d not found in assignments", i)
+					continue
+				}
+				totalAssigned += len(buckets[i])
+
+				// Track which tenants are assigned
+				for _, tenant := range buckets[i] {
+					if assignedTenants[tenant] {
+						t.Errorf("tenant %s assigned to multiple buckets", tenant)
+					}
+					assignedTenants[tenant] = true
+				}
+			}
+
+			// Verify all tenants are assigned
+			if tt.validateAllAssigned && totalAssigned != tt.expectedTenants {
+				t.Errorf("expected %d tenants to be assigned, got %d", tt.expectedTenants, totalAssigned)
+			}
+
+			// Verify each tenant is assigned exactly once
+			for _, tw := range tt.tenantWeights {
+				if !assignedTenants[tw.TenantName] {
+					t.Errorf("tenant %s was not assigned to any bucket", tw.TenantName)
+				}
+			}
+
+			// Verify weights are calculated correctly
+			for i := 0; i < tt.numShards; i++ {
+				calculatedWeight := 0
+				for _, tenantName := range buckets[i] {
+					// Find the weight for this tenant
+					for _, tw := range tt.tenantWeights {
+						if tw.TenantName == tenantName {
+							calculatedWeight += tw.Weight
+							break
+						}
+					}
+				}
+				if weights[i] != calculatedWeight {
+					t.Errorf("bucket %d: expected weight %d, got %d", i, calculatedWeight, weights[i])
+				}
+			}
+		})
+	}
+}
+
+func TestComputeTenantAssignments_DistributionFairness(t *testing.T) {
+	// Test that distribution is relatively fair
+	numShards := 3
+	tenantWeights := []TenantWeight{
+		{TenantName: "tenant1", Weight: 100},
+		{TenantName: "tenant2", Weight: 90},
+		{TenantName: "tenant3", Weight: 80},
+		{TenantName: "tenant4", Weight: 70},
+		{TenantName: "tenant5", Weight: 60},
+		{TenantName: "tenant6", Weight: 50},
+	}
+
```

Comment:
Try to take a loot at other thanos unit test examples, such as https://github.com/databricks/thanos/blob/db_main/pkg/compact/overlapping_test.go#L22 which decouple boilplate code from test data

also try to test when weights are imbalanced, 1 big tenants got assigned sololy to 1 compactor